### PR TITLE
Generic syscall kprobes

### LINKED ISSUE #4211: change socket_dup to not rely on sys_enter/exit

### PR DESCRIPTION
**Pull request by yanivagman**
### 1. Explain what the PR does

feat: Add syscall kprobe support

We currently use sys_enter/sys_exit tracepoints to trace syscalls.
These impact performance on hot paths.

This change introduces BPF program attachment to syscall entry/exit using kprobes,
offering a more efficient mechanism. 
It handles varying symbol prefixes across architectures and includes 32-bit compat support.

### 3. Other comments

Close #4211

### CODE REVIEW

**Comment by yanivagman** [comment]
This PR implements a generic way to add probes for system calls using kprobes.
It replaces PRs #4229 #4231 #4232

File: pkg/ebpf/c/common/context.h [hunk 1]
```diff
@@ -191,6 +191,7 @@ statfunc int init_program_data(program_data_t *p, void *ctx, u32 event_id)
     p->event->config.submit_for_policies = ~0ULL;
 
     if (event_id != NO_EVENT_SUBMIT) {
+        p->event->config.submit_for_policies = 0;
```
Comment (geyslan) [inline pkg/ebpf/c/common/context.h:194]: silent bug?

**Review by geyslan — APPROVED** [review]
LGTM

File: pkg/ebpf/c/common/context.h (hunk 1 above)
Comment (yanivagman) [reply pkg/ebpf/c/common/context.h:194]: yep

**Pull request merged**